ledger/ledgercore/statedelta.go

```diff
@@ -214,6 +223,16 @@ func MakeAccountDeltas(hint int) AccountDeltas {
 	}
 }
 
+// ReuseStateDelta resets fields that aren't directly set by PopulateStateDelta
+func (sd *StateDelta) ReuseStateDelta(hdr *bookkeeping.BlockHeader, prevTimestamp int64, hint int, stateProofNext basics.Round) {
```

Comment:
I suggest kind of real reuse of `Accts` and maps - if you implement/call something like sd.reuse/recycle before returning to a pool that slices the slice to zero length and "clears" the maps.

Reply:
Ok I left it like this because we initially spoke about doing this for child object only. If we are implementing true full re-use of `StateDelta`  would it make sense to implement a separate pool for them as well so that `makeRoundCowState` benefits as well?

Reply:
you are not implementing pool for StateDelta, but reset it state instead that is simpler and should be good enough

Reply:
We are using hints here throughout. Should we only re-use if old cap is >= new hint? to avoid over-allocating when slice expands  or multiple allocs if it has to expand multiple times?